### PR TITLE
limit3.comp: add 'in-limit' output pin

### PR DESCRIPTION
**Pull request by jmkasunich**
The 'in-limit' pin is false when the output is exactly
tracking the input, and true when position, velocity,
or accel limit causes the output to differ from the
input.

Also replaced test limit3.1 (which was a duplicate
of limit3.0) with a more complete test that verifies
accel and velocity limitation and the 'in-limit' pin.

Backwards compatible - just ignore the new output.

Signed-off-by: John Kasunich <jmkasunich@fastmail.fm>

### CODE REVIEW

**Comment by jepler** [comment]
@jmkasunich because of the way github pull requests work, the second commit you added to your master branch, "halstreamer: allow comments in input stream" becomes a part of this pull request.  To have multiple outstanding pull requests with different contents, you need to use a different branch name for each one.  Each branch should start at linuxcnc "uptream" master.

There's also at least one problem with the limit3 change that I spotted, look for the line comment I'll make in just a moment...

File: src/hal/components/limit3.comp [hunk 1]
```diff
@@ -22,19 +23,24 @@ typedef struct {
...
     }
     if ( lin > max_ ) {
 	lin = max_;
+        inlimit = 1;
     }
 
     if(load) {
 	data.old_in = data.old_out = out = lin;
 	data.old_v = 0;
+        in_limit = 1;
```
Comment (jepler) [inline src/hal/components/limit3.comp]: should be an assignment to inlimit, not in_limit here, I think

Comment (jmkasunich) [inline src/hal/components/limit3.comp]: if (load) is a special case - it stuffs the new values, sets in_limit, and returns without doing any of the rest of the code.
If it set the internal variable inlimit, that value would never be copied to the hal pin.

**Comment by jmkasunich** [comment]
On Tue, Aug 16, 2016, at 11:06 AM, Jeff Epler wrote:
> @jmkasunich because of the way github pull requests work, the second commit you added to your master branch, "halstreamer: allow comments in input stream" becomes a part of this pull request.  To have multiple outstanding pull requests with different contents, you need to use a different branch name for each one.  Each branch should start at linuxcnc "uptream" master.
> 
Well that's annoying.  I started a pull request for the second commit
last night, when I saw that it was including the first commit (which 
hadn't been merged yet), I stopped, figuring I would re-submit after
the first one merged.  I didn't realize that it added the second commit
to the first PR, even though the commit happened AFTER the PR.  
That's just dumb IMO.

I guess in the future I'll have to use branches.  Although in this case 
could I have submitted a single PR for all four commits?  Each commit
stands alone and has its own meaningful commit message, but does
putting them all in one pull request mess anything up?  Seems like it
would be less work for the maintainers.

> There's also at least one problem with the limit3 change that I spotted, look for the line comment I'll make in just a moment...

I answered that separately (from github).  The "if (load)" branch 
returns without running the rest of the code, so I need to set the
hal pin, not the internal variable.

-- 
  John Kasunich
  jmkasunich@fastmail.fm

File: src/hal/components/limit3.comp (hunk 1 above)
Comment (jepler) [inline src/hal/components/limit3.comp]: I see that now, thanks for explaining.

**Pull request merged**